This thesis highlights experimental and computational approaches to understanding cellular metabolism.
Chapter \ref{chapter:A} demonstrates application of transcriptomics to a pure Type II methanotroph culture growing under methane-grown conditions.
Strong conclusions about which metabolic pathways are active were possible given the high-quality reference genome available for RNA-seq alignments.

Chapter \ref{chapter:B} extends these techniques to species rich microbial communities, and uses time-series and experimental differences to probe the driving factors for community composition.
In this case, the only reference DNA available for alignments was a handful of isolate genomes, and contigs assembled from the pooled 88 metagenomes.
The ability to derive high-quality inferences about the genetic potential and gene expression levels of these samples is much lower than for the pure culture case of Chapter \ref{chapter:A}.
A high-level understanding of which microbes dominate the sample is provided, and steps to refine understanding are outlined.

The potential of extracting additional information from the gene-expression levels found in Chapter \ref{chapter:B} is explored in Chapter \ref{chapter:C}.
The potential of canonical correlation analysis to identify correlations in the gene-expression data is tested on a preliminary data set, and shrinkage-based estimates of the partial correlation matrix were used to identify cross-species gene pairs that co-occur across samples.
Metrics for validating such partial correlation networks are proposed, and specific aims for mining these networks for between-species interactions are proposed.

In total, this work demonstrates use of wet-lab and computational approaches to gain the type of knowledge necessary to engineer microbes.
I am thrilled to continue pushing the boundaries of sustainable chemical production using my combination of expertise in microbiology, wet-lab experimentation, and data science.

