\chapter*{Glossary} % \chapter*{Glossary} <-- the star turns off the chapter numbering and removes it from the index.
\renewcommand*{\arraystretch}{1.4} % space between entries % http://tex.stackexchange.com/questions/135593/adding-row-spacing-to-a-longtable

\begin{singlespace}
\begin{longtable}{ p{.20\textwidth}  p{.80\textwidth} }

\textbf{AMI} & Amazon machine image.  A master image for the creation of virtual servers (known as EC2 instances)
                in the Amazon Web Services (AWS) environment. \\
\textbf{ANI} & Average nucleotide identity, a measure of similarity between organisms. \\
\textbf{assembly} &  The process of, or result of inferring the original genome sequences that produced sequencing reads. \\
\textbf{AWS} & Amazon Web Services, the leading cloud computing platform. \\
\textbf{AWS instance} & A single computer rented from Amazon Web Services, with user-selected performance characteristics. \\
\textbf{bin} & A collection of contigs that approximates one organism, or a group of closely related strains. \\
\textbf{binning} &  The process of clustering contigs into bins.  \\ % "clustering"?
\textbf{BWA} & Burrows-Wheeler Aligner, a tool for mapping reads to reference DNA sequences. \\
\textbf{CCA} & Canonical correlation analysis. \\
\textbf{CheckM} &  A computational tool used to assess genome bin completeness and contamination. \\
\textbf{contig} & A contiguous DNA sequence.  In this case, they are the longer DNA stretches identified from assembling Illumina reads. \\
\textbf{C1} & Single-carbon compounds such as methane, methanol, and formate. \\
\textbf{coverage} & The average number of times a base of a genome (or genome fragment) is sequenced.  \\
\textbf{cross-validation} &  A model validation technique that loops over subsets of the data, and leads to an assessment of how the results of a model will generalize to an independent data set.  \\
\textbf{d} & The dimensionality of the data.  In this thesis, it is usually the number of genes. \\
\textbf{EC2} & A virtual computing environment within Amazon Web Services. \\
\textbf{EFS} & Amazon's Elastic File System, a scalable file storage service.  \\
\textbf{Elviz} & A Joint Genome Analysis toolchain for metagenomics and metatranscriptomics.  \\
\textbf{FASTA} &  A biology sequence data format, which can contain DNA, RNA, or protein sequences for different units of information such as genes, contigs, or whole genomes.  Individual FASTA files are referred to as \texttt{.fasta} files herein.  \\
\textbf{fastq} & A \texttt{.fasta}-like file with associated sequence quality information.  This is the most raw output from the sequencing platform handled in this thesis. \\
\textbf{feature} & A single measurable property of a phenomenon being observed, such as one gene's expression level. \\
\textbf{Gaussian distribution} & Normal distribution; a commonly used symmetric distribution. \\
\textbf{GC} & Strictly speaking, G and C are each DNA bases.
	When used in terms like "high GC", it indicates high fraction of G, C nucleotides over A, T nucleotides. \\
\textbf{genomics} & The collective characterization and quantification of pools of DNA molecules \\
\textbf{genome bin} & A collection of contigs used to approximate a single organism's genome.  Often referred to as "bin". \\
%\textbf{graph} &  \\
%\textbf{held-out set} &  \\
\textbf{hyperparameter} & Parameters in a machine learning model that express higher-level properties of the model such as the model complexity or the learning rate. \\
\textbf{i.i.d.} & Independent and identically distributed.
    Each random variable has the same probability distribution as the others and all are mutually independent. \\
\textbf{isolate} & A single microbial strain that has been isolated and propagated in pure culture. \\
\textbf{JGI} & The Joint Genome Institute, a U.S. Department of Energy laboratory which provided sequencing
                services for the metagenomics work of this thesis. \\
\textbf{kb} & Kilobase. A unit of DNA (or RNA) length, corresponding to 1000 base pairs. \\
\textbf{L1 regularization} & A method to limit model complexity by penalizing the absolute value of model coefficients.
     Leads to sparse models (truly zero coefficients), rather than coefficients with small magnitudes (e.g. 0.00001). \\
\textbf{mapping} & Alignment of sequencing reads to target DNA such as genome or contig sequences.\\
\textbf{machine learning} & Models that have the ability to learn without being explicitly programmed. \\
\textbf{MetaBAT} &  A binning tool. \\
\textbf{metagenome} & DNA recovered directly from a mixed microbial community, rather than an isolated strain. \\
\textbf{metatranscriptome} & RNA recovered directly from a mixed microbial community, rather than an isolated strain. \\
\textbf{methanotroph} & A microbe which can use methane as its sole carbon and energy source. \\
\textbf{methylotroph} & A microbe which can use reduced C1 compounds such as methanol as its sole carbon and energy source. \\
\textbf{methane monooxygenase} & (MMO) A methanotroph enzyme that can oxidize the C-H bond in preparation for assimilation or energy production. \\
\textbf{multilocus mapped reads} & (MMR) Reads that map more than one place equally well, causing uncertainty about their true origin. \\
\textbf{non-methanotrophic methylotroph} & While all methanotrophs are methylotrophs, not all methylotrophs can utilize methane.
	Such methylotrophs are often described as non-methanotrophic methylotrophs. \\
\textbf{MyCC} & A binning tool. \\
\textbf{N} & The number of samples available for machine learning, e.g. 88 for 88 metatranscriptomes. \\
%\textbf{N50} & A weighted median statistic for assemblies.  %http://biology.stackexchange.com/questions/34122/an-example-for-n50-why-do-we-need-it
%    Defined such that 50\% of the entire assembly is contained in contigs or scaffolds equal to or larger than this value. \\
    %the length for which the collection of all contigs of that length, or longer, contains at least half of the total of the
    %            lengths of the contigs in the assemblyâ€ \\ % https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4426941/pdf/bbi-9-2015-075.pdf (cites 49, 50)
\textbf{OB3b} & \textit{Methylosinus trichosporium} {OB3b}, a RuMP cycle methanotroph, that is the subject of study in Chapter \ref{chapter:A}  \\
\textbf{omics} & Refers to fields of study ending in \textit{-omics}, such as genomics and transcriptomics, which quantify pools of biological molecules. \\
\textbf{partial correlation} & A measure of the strength and direction of a linear relationship between two continuous variables whilst controlling for the effect of one or more other continuous variables. \\
\textbf{PCA} & Principle components analysis. \\
\textbf{PhyloPhlAn} & A computational tool that can assign taxonomy to metagenome bins. \\
\textbf{Prokka} & A computational tool for annotating genes on contigs. \\
\textbf{read} & a single short DNA sequence of a larger high-throughput sequencing data set. \\
\textbf{regularization} & A technique to limit model complexity and prevent over-fitting to training data. \\
\textbf{RPKM} & Reads per kilobase per kilobase mapped. \\ % \cite{mortazavi2008} \\
\textbf{sparse} & Data or computational models with many zero values or coefficients. \\
\textbf{taxa} & A taxonomic category, such as a genus or species. \\
\textbf{taxonomy} & A scheme of classification for organisms. \\
\textbf{tetranucleotide frequency} & The frequency of 4-nucleotide sequences in DNA, which is conserved for an organism. \\
\textbf{TPM} & %Transcripts per million \cite{wagner2012}, an alternative to RPKM \cite{mortazavi2008} \\
	Transcripts Per Kilobase Million, an alternative to RPKM.
	Reads per kilobase is calculated first, then these totals are normalized by dividing by the sum and 1,000,000. \\
	% Divide the read counts by the length of each gene in kilobases. This gives you reads per kilobase (RPK).
	% Count up all the RPK values in a sample and divide this number by 1,000,000. This is your ?per million? scaling factor.
	% Divide the RPK values by the ?per million? scaling factor. This gives you TPM.
\textbf{training set} &  Data used to train a machine learning model. \\
\textbf{transcriptomics} & The collective characterization and quantification of pools of RNA molecules (transcripts). \\
\textbf{16S} & A ribosomal protein, whose DNA sequence is commonly used for evolutionary inference. \\

\end{longtable}
\end{singlespace}
